pkg/odo/cli/component/push.go

```diff
@@ -250,13 +251,25 @@ func (po *PushOptions) Validate() (err error) {
 
 // Run has the logic to perform the required actions as part of command
 func (po *PushOptions) Run(cmd *cobra.Command) (err error) {
+	scontext.SetClusterType(cmd.Context(), po.Client)
 	// If experimental mode is enabled, use devfile push
 	if util.CheckPathExists(po.DevfilePath) {
+		var componentType string
+		metadata := po.Devfile.Data.GetMetadata()
+		if metadata.ProjectType != "" {
+			componentType = metadata.ProjectType
+		} else if metadata.Language != "" {
+			componentType = metadata.Language
+		} else {
+			componentType = metadata.Name
+		}
+		scontext.SetComponentType(cmd.Context(), componentType)
```

Comment:
This code block is repeating in https://github.com/openshift/odo/pull/4763/files#diff-e9cbb4b32525048f3bf6ffce25bb7e24c02221819d5b9de9347fe2c835828058R841-R850. Can we extract into a function?